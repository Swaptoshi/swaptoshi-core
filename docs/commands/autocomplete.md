# `swaptoshi-core autocomplete`

display autocomplete installation instructions

- [`swaptoshi-core autocomplete [SHELL]`](#swaptoshi-core-autocomplete-shell)

## `swaptoshi-core autocomplete [SHELL]`

display autocomplete installation instructions

```
USAGE
  $ swaptoshi-core autocomplete [SHELL] [-r]

ARGUMENTS
  SHELL  shell type

FLAGS
  -r, --refresh-cache  Refresh cache (ignores displaying instructions)

DESCRIPTION
  display autocomplete installation instructions

EXAMPLES
  $ swaptoshi-core autocomplete

  $ swaptoshi-core autocomplete bash

  $ swaptoshi-core autocomplete zsh

  $ swaptoshi-core autocomplete --refresh-cache
```

_See code: [@oclif/plugin-autocomplete](https://github.com/oclif/plugin-autocomplete/blob/v1.3.6/src/commands/autocomplete/index.ts)_
