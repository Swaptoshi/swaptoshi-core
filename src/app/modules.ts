/* eslint-disable @typescript-eslint/no-empty-function */
import { Application, Modules } from 'klayr-sdk';
import { Modules as SwaptoshiModules } from 'swaptoshi-sdk';

interface KlayrMethod {
	validator: Modules.Validators.ValidatorsMethod;
	auth: Modules.Auth.AuthMethod;
	token: Modules.Token.TokenMethod;
	fee: Modules.Fee.FeeMethod;
	random: Modules.Random.RandomMethod;
	reward: Modules.DynamicReward.DynamicRewardMethod;
	pos: Modules.PoS.PoSMethod;
	interoperability: Modules.Interoperability.SidechainInteroperabilityMethod | Modules.Interoperability.MainchainInteroperabilityMethod;
}

export const registerModules = (app: Application, method: KlayrMethod): void => {
	const nftModule = new Modules.NFT.NFTModule();
	const tokenFactoryModule = new SwaptoshiModules.TokenFactory.TokenFactoryModule();
	const dexModule = new SwaptoshiModules.DEX.DexModule();
	const feeConversionModule = new SwaptoshiModules.FeeConversion.FeeConversionModule();
	const liquidPosModule = new SwaptoshiModules.LiquidPoS.LiquidPosModule();
	const governanceModule = new SwaptoshiModules.Governance.GovernanceModule();

	nftModule.addDependencies(method.interoperability, method.fee, method.token);
	governanceModule.addDependencies({ tokenMethod: method.token, feeMethod: method.fee });
	liquidPosModule.addDependencies({ tokenMethod: method.token, governanceMethod: governanceModule.method });
	feeConversionModule.addDependencies({ tokenMethod: method.token, feeMethod: method.fee, dexMethod: dexModule.method, governanceMethod: governanceModule.method });
	dexModule.addDependencies({
		tokenMethod: method.token,
		nftMethod: nftModule.method,
		feeMethod: method.fee,
		interoperabilityMethod: method.interoperability,
		feeConversionMethod: feeConversionModule.method,
		governanceMethod: governanceModule.method,
	});
	tokenFactoryModule.addDependencies({
		tokenMethod: method.token,
		feeMethod: method.fee,
		nftMethod: nftModule.method,
		interoperabilityMethod: method.interoperability,
		dexMethod: dexModule.method,
		feeConversionMethod: feeConversionModule.method,
		governanceMethod: governanceModule.method,
	});

	// governance module needs to executed before dynamicReward module for treasury blockRewardTax
	app.registerModulePriority(governanceModule);

	// feeConversion module needs to be executed before fee module for altering fee verification, and before all modules that implements fc_method
	app.registerModulePriority(feeConversionModule);

	app.registerModule(nftModule);
	app.registerModule(tokenFactoryModule);
	app.registerModule(dexModule);
	app.registerModule(liquidPosModule);

	// above registration order will result in following module array:
	// [feeConversion, governance, ...klayrModules, nftModule, tokenFactoryModule, dexModule, liquidPosModule]
	// the later the registerModulePriority, the higher the priority (unshift)

	app.registerInteroperableModule(nftModule);
	app.registerInteroperableModule(tokenFactoryModule);
	app.registerInteroperableModule(dexModule);
};
