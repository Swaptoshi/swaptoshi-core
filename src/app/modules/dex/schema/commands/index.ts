export * from './burn_command';
export * from './collect_command';
export * from './collect_treasury_command';
export * from './create_pool_command';
export * from './decrease_liquidity_command';
export * from './exact_input_command';
export * from './exact_input_single_command';
export * from './exact_output_command';
export * from './exact_output_single_command';
export * from './increase_liquidity_command';
export * from './mint_command';
export * from './treasurify_command';
