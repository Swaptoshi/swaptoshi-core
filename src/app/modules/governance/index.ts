export { GovernanceModule } from './module';
export { GovernanceMethod } from './method';
export { BaseGovernableConfig } from './base_governable_config';
export * from './types';
