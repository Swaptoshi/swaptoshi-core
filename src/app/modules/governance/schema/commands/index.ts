export * from './create_proposal_command';
export * from './vote_command';
export * from './boost_vote_command';
export * from './delegate_vote_command';
export * from './revoke_delegated_vote_command';
export * from './set_proposal_attributes';
