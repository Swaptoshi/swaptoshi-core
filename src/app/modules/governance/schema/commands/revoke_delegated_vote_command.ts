import { RevokeDelegatedVoteParams, TypedSchema } from '../../types';

export const revokeDelegatedVoteCommandSchema: TypedSchema<RevokeDelegatedVoteParams> = {
	$id: '/governance/command/revoke_delegated_vote',
	type: 'object',
	required: [],
	properties: {},
};
