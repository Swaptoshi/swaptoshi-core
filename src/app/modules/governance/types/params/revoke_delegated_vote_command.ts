export type RevokeDelegatedVoteParams = Record<string, never>;
