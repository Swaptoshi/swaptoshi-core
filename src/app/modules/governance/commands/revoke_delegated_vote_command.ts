/* eslint-disable */
import { BaseCommand, CommandVerifyContext, CommandExecuteContext, VerificationResult, VerifyStatus } from 'klayr-sdk';
import { RevokeDelegatedVoteParams } from '../types';
import { immutableTransactionHookGovernanceContext, mutableTransactionHookGovernanceContext } from '../stores/context';
import { DelegatedVoteStore } from '../stores/delegated_vote';
import { revokeDelegatedVoteCommandSchema } from '../schema';

export class RevokeDelegatedVoteCommand extends BaseCommand {
	public async verify(_context: CommandVerifyContext<RevokeDelegatedVoteParams>): Promise<VerificationResult> {
		try {
			const context = immutableTransactionHookGovernanceContext(_context);
			const delegatedVote = await this.stores.get(DelegatedVoteStore).getImmutableDelegatedVote(context);
			await delegatedVote.verifyRevokeDelegatedVote(_context.params);
		} catch (error: unknown) {
			return {
				status: VerifyStatus.FAIL,
				error: new Error((error as { message: string }).message),
			};
		}
		return { status: VerifyStatus.OK };
	}

	public async execute(_context: CommandExecuteContext<RevokeDelegatedVoteParams>): Promise<void> {
		const context = mutableTransactionHookGovernanceContext(_context);
		const delegatedVote = await this.stores.get(DelegatedVoteStore).getMutableDelegatedVote(context);
		await delegatedVote.revokeDelegatedVote(_context.params, false);
	}

	public schema = revokeDelegatedVoteCommandSchema;
}
