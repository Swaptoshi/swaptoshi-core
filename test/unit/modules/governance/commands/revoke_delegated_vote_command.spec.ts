/* eslint-disable */
import { GovernanceModule } from '../../../../../src/app/modules/governance';
import { RevokeDelegatedVoteCommand } from '../../../../../src/app/modules/governance/commands/revoke_delegated_vote_command';

describe('RevokeDelegatedVoteCommand', () => {
	let module: GovernanceModule;
	let command: RevokeDelegatedVoteCommand;

	beforeEach(() => {
		module = new GovernanceModule();
		command = new RevokeDelegatedVoteCommand(module.stores, module.events);
	});

	describe('constructor', () => {
		it('should have valid name', () => {
			expect(command.name).toEqual('revokeDelegatedVote');
		});

		it('should have valid schema', () => {
			expect(command.schema).toMatchSnapshot();
		});
	});

	describe('verify', () => {
		describe('schema validation', () => {
			it.todo('should throw errors for invalid schema');
			it.todo('should be ok for valid schema');
		});
	});

	describe('execute', () => {
		describe('valid cases', () => {
			it.todo('should update the state store');
		});

		describe('invalid cases', () => {
			it.todo('should throw error');
		});
	});
});
