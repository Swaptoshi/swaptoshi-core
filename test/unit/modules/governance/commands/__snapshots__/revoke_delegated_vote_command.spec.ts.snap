// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RevokeDelegatedVoteCommand constructor should have valid schema 1`] = `
{
  "$id": "/governance/command/revoke_delegated_vote",
  "properties": {},
  "required": [],
  "type": "object",
}
`;
